File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] arr, Integer target) {
        int[] ans=new int[2];
        ans[0]=-1;
        ans[1]=-1;
       int n=arr.length;
        int sum=0;
        HashMap<Integer,Integer>map=new HashMap<>();
        for(int k=0;k<n;k++)
        {
        map.put(arr[k],k);    
        }
      for(int i=0;i<arr.length;i++)
      {
          int dif=target-arr[i];
          if(map.containsKey(dif))
          {
              if(map.get(target-arr[i])!=i)
              {
             ans[0]=i;
              ans[1]=map.get(target-arr[i]);
              break; 
              }
             
          }
      }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], k: number): number[] {
    let ans:number[]=[]
    let map=new Map<number,number>
    let n=nums.length
    for(let i=0;i<n;i++){
        if(map.has(k-nums[i])){
            ans[0]=i
            ans[1]=map.get(k-nums[i])
            return ans
        }
        map.set(nums[i],i)
    }
    return ans
};